### PR TITLE
feat: Change auth template to match spec

### PR DESCRIPTION
**Pull request by david-code**
## Description

Adds a test to make sure the SSO authorization template is being rendered successfully. Also adjusts the wording to match the specifications.

### CODE REVIEW

File: terraso_backend/tests/auth/test_templates.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from django.urls import reverse
+from oauth2_provider.models import Application
+
+pytestmark = pytest.mark.django_db
+
+
+def test_authorization_template(client, user):
+    application = Application(
+        name="Test Application",
+        redirect_uris=("http://example.org"),
```
Comment (paulschreiber) [inline terraso_backend/tests/auth/test_templates.py:26]:
```suggestion
        redirect_uris=("https://example.org"),
```

File: terraso_backend/tests/auth/test_templates.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        authorization_grant_type=Application.GRANT_AUTHORIZATION_CODE,
+        algorithm=Application.RS256_ALGORITHM,
+        client_secret="1234567890abcdefghijklmnopqrstuvwxyz",
+    )
+    application.save()
+    query_data = {
+        "client_id": application.client_id,
+        "scope": "openid email",
+        "state": "random_state_string",
+        "redirect_uri": "http://example.org",
```
Comment (paulschreiber) [inline terraso_backend/tests/auth/test_templates.py:38]:
```suggestion
        "redirect_uri": "https://example.org",
```

File: terraso_backend/custom_templates/oauth2_provider/authorize.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+along with this program. If not, see https://www.gnu.org/licenses/.
+-->
+{% extends "oauth2_provider/base.html" %}
+
+{% load i18n %}
+{% block content %}
+    <div class="block-center">
+        {% if not error %}
+            <form id="authorizationForm" method="post">
+              <!-- <h3 class="block-center-heading">{% trans "Authorize" %} {{ application.name }}?</h3> -->
```
Comment (paulschreiber) [inline terraso_backend/custom_templates/oauth2_provider/authorize.html:24]: suggestions: uncomment or remove this line.

File: terraso_backend/custom_templates/oauth2_provider/authorize.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                <h3 class="block-center-heading">{% trans "Share your name and email with " %} {{ application.name}} ?</h3>
```
Comment (paulschreiber) [inline terraso_backend/custom_templates/oauth2_provider/authorize.html:25]:
```suggestion
                <h3 class="block-center-heading">{% trans "Share your name and email with" %} {{ application.name}} ?</h3>
```

File: terraso_backend/custom_templates/oauth2_provider/authorize.html (hunk 3 above)
Comment (david-code) [inline terraso_backend/custom_templates/oauth2_provider/authorize.html:24]:
```suggestion
            <form id="authorizationForm" method="post">
```

**Review by paulschreiber — APPROVED** [review]

**Comment by paulschreiber** [comment]
@david-code please fix the test failure before merging.

**Pull request merged**